src/main/java/com/microsoft/sqlserver/jdbc/SQLServerPreparedStatement.java

```diff
@@ -2548,6 +2647,326 @@ public final void addBatch() throws SQLServerException {
         loggerExternal.exiting(getClassNameLogging(), "executeLargeBatch", updateCounts);
         return updateCounts;
     }
+    
+    
+    private String parseUserSQLForTableNameDW(boolean hasInsertBeenFound, boolean hasIntoBeenFound) {
+        // As far as finding the table name goes, There are two cases:
+        // Insert into <tableName> and Insert <tableName>
+        // And there could be in-line comments (with /* and */) in between.
+        // This method assumes the localUserSQL string starts with "insert".
+        localUserSQL = localUserSQL.trim();
+        if (checkAndRemoveComments()) {
```

Comment:
Don't you think there are too many unnecessary calls(these methods are all recursive) to `checkAndRemoveComments()` assuming most of the queries will contain no comments at all? Would it be easier/faster just to traverse the string at the very beginning and remove all comments?

Reply:
Could be, but SQL Server treats the comment as a blank space. For example, the query below:

insert into t1/*this is a comment*/values (1);

will execute normally as if there was a space in between 't1' and 'values'. This means that if we were to sweep over the user's SQL query and remove all the comments, we would have to replace all the comments with a space, which would alter the user's SQL (that we will use to parse for the table names and such), which I don't think is better than the current approach.